hypershift-operator/controllers/hostedclustersizing/hostedclustersizing_controller.go

```diff
@@ -216,7 +217,9 @@ func (r *reconciler) reconcile(
 
 	// first, we figure out the node count for the hosted cluster
 	var nodeCount uint32
+	nodeCountRequiresAPIServer := false
```

Comment:
Can we include a comment for future code readers? e.g. "When the node count comes from status.replicas we only trust it if the kas is available, otherwise we might get false 0s, https://github.com/kubernetes-sigs/cluster-api/issues/10195".